compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSDeclarationImpl.kt

```diff
@@ -44,7 +46,15 @@ abstract class KSDeclarationImpl(ktDeclaration: KtDeclaration) : KSDeclaration {
     }
 
     override val modifiers: Set<Modifier> by lazy {
-        ktDeclaration.toKSModifiers()
+        val modifiers = ktDeclaration.toKSModifiers()
+        val hasJvmStatic = annotations.any {
```

Comment:
`@JvmStatic` can only be applied to function and property and property accessors, consider check for type to avoid annotation type resolution on class declarations.

Reply:
Also consider check for `referencedName()` before calling resolution.

Reply:
how do i get `referencedName` ?
I can use `shortName.asString()` but i'm also afraid it won't cover type aliases. (which i can try :) )

Reply:
hmm actually, when typealiases are used, resolve returns `KSTypeImpl` with [`typealias CustomJvmStatic`] and its declaration returns `KSTypeAliasImpl` to that typealias declaration.

this is probably expected but it might be a land mine for developers.

Created https://issuetracker.google.com/issues/184399156 to ensure we handle this properly.

Reply:
i noticed we have a `findActualType` method to handle these.
But i have to resolve to get there so i think i cannot do the name check here.

Reply:
OK i added the check for Kt types and also tests w/ type aliases.